kubernetes/skaffold/src/main/kotlin/com/google/kubernetes/tools/skaffold/run/SkaffoldRunConfigurations.kt

```diff
@@ -66,10 +69,22 @@ class SkaffoldDevConfiguration(
 ) : AbstractSkaffoldRunConfiguration(project, factory, name) {
 
     override fun getState(executor: Executor, environment: ExecutionEnvironment): RunProfileState? =
-        SkaffoldCommandLineState(environment, SkaffoldExecutorSettings.ExecutionMode.DEV)
+        SkaffoldCommandLineState(environment, getExecutionMode(environment))
 
     override fun getConfigurationEditor(): SettingsEditor<out RunConfiguration> =
         SkaffoldDevSettingsEditor()
+
+    /**
+     * Returns the execution mode depending on the execution environment, i.e. if it was run in
+     * either 'run' or 'debug' mode.
+     */
+    @VisibleForTesting
+    fun getExecutionMode(environment: ExecutionEnvironment) = when (environment.executor) {
+            is DefaultRunExecutor -> SkaffoldExecutorSettings.ExecutionMode.DEV
+            is DefaultDebugExecutor -> SkaffoldExecutorSettings.ExecutionMode.DEBUG
+            else -> throw RuntimeException("Unexpected Skaffold executor type found: " +
```

Comment:
IDE APIs change time to time, what if just return `DEV` for all other executors instead of accepting strictly `DefaultRunExecutor`?

Reply:
It also can vary slightly across non-Java IDEs (don't have any hard evidence though) - from previous experience with APIs

Reply:
that sgtm. will update

Reply:
updated